xbmc/filesystem/ShoutcastFile.cpp

```diff
@@ -152,14 +153,23 @@ bool CShoutcastFile::ExtractTagInfo(const char* buf)
   g_charsetConverter.wToUTF8(wConverted, strBuffer);
 
   CRegExp reTitle(true);
-  reTitle.RegComp("StreamTitle=\'(.*?)\';");
+  reTitle.RegComp("StreamTitle=\'(.+?)\';");
 
   if (reTitle.RegFind(strBuffer.c_str()) != -1)
   {
     std::string newtitle(reTitle.GetMatch(1));
     CSingleLock lock(m_tagSection);
-    result = (m_tag.GetTitle() != newtitle);
-    m_tag.SetTitle(newtitle);
+    result = !StringUtils::EqualsNoCase(newtitle, m_tag.GetTitle());
+    if (result)
+    {
+      if ( std::find_if(newtitle.begin(), newtitle.end(), ::islower) == newtitle.end())
+      {
+        // use title case with all caps text
```

Comment:
Maybe change the comment slightly to something like.

`// use title case if text is all caps`

As title is also in the variable name I think this makes it a little clearer

Reply:
Changed